### DISCUSSION TITLE
Better support for block themes and the `Your homepage displays` option

### DISCUSSION THREAD

**Discussion by felixarntz** [WordPress/gutenberg#64620] [Developer Experience]
Almost a year ago, we were discussing how to handle the `home` and `front-page` templates in the TT4 theme (see https://github.com/WordPress/twentytwentyfour/pull/706). This highlighted a problem where block themes are unable to correctly honor the `Your homepage displays` WordPress option, because the `front-page` template is loaded regardless of the option value. At the same time, the theme has no influence on that option, which can lead to a poor user experience, since many themes are either focused on being for a blog or for a site where the blog is _not_ the primary content.

So I believe there are two underlying problems:
1. Block themes are currently unable to adequately honor the user choice of whether the site has a "static front page" or "latest posts". This is not a theme problem, but a core problem and it's what https://core.trac.wordpress.org/ticket/59767 is for. I just commented on that ticket earlier, and I think it would be great to address it now that TT5 is around the corner and will likely run into the same problem.
2. The initial UX of starting with a new theme is poor, if WordPress's front page setting is configured in a way that the theme doesn't "prefer". There are themes intended for a blog front page and there are themes intended for a static front page. It would be great if a theme could specify that preference in its code, and WordPress would react accordingly. I found an older ticket that sort of describes this idea (although I would make some changes to the exact proposed implementation): https://core.trac.wordpress.org/ticket/19627

In other words, it goes both ways. Block themes need to be able to support the way in which the option is configured, but at the same time the option should better support the theme's preference.

Separately, it's worth noting that, per an HTTP Archive query I ran earlier in https://github.com/GoogleChromeLabs/wpp-research/pull/150, almost 80% of WordPress home pages display a static page. So potentially showing the blog on the home page is just not a good default nowadays in the first place. But that's an even broader discussion, so maybe better to focus on the two problems above that directly affect themes. This information may just be useful context.

Curious to hear other people's thoughts on this. How can we improve the compatibility between WordPress (block) themes and the `Your homepage displays` option?

**Comment by luminuu** [comment]
Pinging TT5 leads: @carolinan @juanfra 
Also pinging this groups for visibility, as we need more eyes on this topic: @WordPress/block-themers @WordPress/gutenberg @WordPress/gutenberg-design @WordPress/outreach

**Comment by mtias** [comment]
This is a thorny area that travels with a lot baggage and causes a lot of frictions to users. It'd be great to take a broad look at the current landscape—including homepage settings, their relationships with permalinks and theme hierarchy, the representation as templates or pages, and how we could aim to simplify the matrix of combinations they produce. We should also aim to simplify without making things simplistic! There are valid reasons for the existence of these combinations and how their dynamics affect the url structure of a website.

It expands beyond the dynamics of template loading as well and we should consider how these templates are discovered and represented in the admin UI. For example, a "blog" can be perceived as a page (or dynamic page if you will, with subpages for archives, etc), yet WordPress doesn't show the blog templates when it lists pages. That made sense when they were uneditable and driven purely by theme code. However, block themes have opened the door for those dynamic pages to be editable, which means their discovery and presence should be taken into account as well. This also naturally extends to custom post types, taxonomies, and their template relationship...

The proposal for `front-page-home` and `front-page-static` is interesting, but before we go there I think we need to consider how we present these to users when _all exist on a theme_ beyond setup, because if we end up with a combination of `home`, `font-page`, `font-page-home`, `front-page-static`—plus the `page` you might have assigned as the static page—a user may end up throughly confused and paralyzed in trying to understand what drives what behavior.

**Comment by felixarntz** [comment]
@mtias 
> This is a thorny area that travels with a lot baggage and causes a lot of frictions to users.

100%. I believe the way it currently works is outdated partially due to WordPress's blogging origins, but also because that option is so decoupled from the current theme.

> It expands beyond the dynamics of template loading as well and we should consider how these templates are discovered and represented in the admin UI.

I'd say there are short-term and long-term wins to achieve here. For instance, the 1st problem I outlined in the description is a bug more than a UX/DX improvement. While we can (and probably should!) change the way this configuration works and is presented to end users, the lack of being able to correctly support the current value in the block theme is something more pressing due to the broken behavior today. That said, I agree we need to consider some of the other points you mention too and implement a fix to that bug in a way that doesn't get in the way of what we want this to look like long-term.

> The proposal for `front-page-home` and `front-page-static` is interesting, but before we go there I think we need to consider how we present these to users when _all exist on a theme_ beyond setup, because if we end up with a combination of `home`, `font-page`, `font-page-home`, `front-page-static`—plus the `page` you might have assigned as the static page—a user may end up throughly confused and paralyzed in trying to understand what drives what behavior.

Great point on the `static` portion, it's kind of ambiguous: Historically it always meant the contents of the assigned page will be shown, but nowadays it's very often the custom block template contents from the relevant theme template. Speaking in today's terminology, in the latter case, you could technically choose to have a "static front page" without assigning a page to it. So based on this there are 3 possible scenarios for the front page today, "blog", "static page content", and "static template content".

I'd say out of the templates you mention, `home` doesn't really belong to them, as that template is meant for the blog (that naming is so confusing!), independently of whether it's also the home page or not. For sites that have the blog as the home page, `front-page` is used _over_ `home`, so that means in terms of templates for the home page, `home` is not important unless the theme doesn't even have any `front-page` template.

**Reply by mtias** [reply]
Yes, for sure, if we can extract some short-term improvements we should do it. It has proven to be difficult to gradually improve the situation without introducing more confusion, though. There's a lot of themes without `front-page` that use `home` for a look that sits in-between a blog and a static home page, for example.

**Reply by felixarntz** [reply]
What I'm proposing in https://core.trac.wordpress.org/ticket/59767 wouldn't change the user experience in any way, the change would only concern theme developers. Worth noting that this fixes a bug more than anything else. The current reality is that you can either have a static page or a blog configured as the front page, but block themes are unable to handle that because of a lack of template granularity.

**Reply by mtias** [reply]
I mean in the sense that these templates would show up in the modal when users create a new one—they get access to the same theme hierarchy. So it wouldn't just be a theme developer concern but would also see expression in the user interface. Not to discount the idea, just adding the context that what is already somewhat confusing can become even more.

**Reply by jameskoster** [reply]
@felixarntz curious to get your thoughts on the discussion in this issue: https://github.com/WordPress/gutenberg/issues/60772

**Reply by felixarntz** [reply]
@mtias That's a good point, I didn't consider how those templates would be exposed in the site editor UI.

@jameskoster Thanks, I replied in there.

On that note, if we proceed with something like #60772, where `show_on_front` would always be `page`, we might be able to go without any new templates. It could work like this:
* The `front-page` template would need to embed the content of the static page assigned as `page_on_front`. It could either be just that page's content (plus header and footer), or there could be additional template content around it, depending on the theme authors' preference.
* The actual page content would still need to be somehow conditional though: If the page is also assigned as the `page_for_posts`, it would need to render the blog (instead of, or in addition to the page's own content).

**Comment by jameskoster** [comment]
It needs some organisation but https://github.com/WordPress/gutenberg/issues/63783 captures many of the issues relating to homepage setting/editing.
